sky/jobs/state.py

```diff
@@ -11,6 +11,7 @@
 
```

Comment:
minor: It seems we have a pretty clear allowed state transition diagram. Could we quickly draw a diagram of the state machine and leave it here: https://github.com/skypilot-org/skypilot/blob/master/sky/jobs/README.md

Reply:
Yes. Also for scheduler. Can follow up in another PR.
Actually I didn't realize we had that README lol